src/mixed.js

```diff
@@ -172,6 +174,8 @@ SchemaType.prototype = {
   },
 
   cast(value, options = {}) {
+    options.value = options.value || value;
+    options.path = options.path || this._path;
```

Comment:
this. shouldn't mutate options...couldn't `this._path` be wrong here, if a schema is reused in multiple different other schemas?